text/0227-separate-semantic-conventions.md

```diff
@@ -0,0 +1,179 @@
+in the same file), then this can be refactored in the new repository, preserving
+git history.
+
+There will also be the following exceptions in the specification:
+
+- Semantic conventions used to implement API/SDK details will be fully specified in the `opentelemetry-specification` repo
+  and will not be allowed to change in the Semantic Convention directory.
+  - Error/Exception handling will remain in the specification.
+  - SDK configuration interaction w/ semantic convention will remain in the
+    specification. Specifically `service.name`.
+- The specification may elevate some semantic conventions as necessary for
+  compatibility requirements, e.g. `service.instance.id` and
+  [Prometheus Compatibility](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/compatibility/prometheus_and_openmetrics.md).
+
+These exceptions exist because:
+
+- Stable portions of the specification already rely on these conventions.
+- These conventions are required to implement an SDK today.
+
+As such, the Specification should define the absolute minimum of reserved or
+required attribute names and their interaction with the SDK.
+
+## Internal details
+
+The following process would be used to ensure semantic conventions are
+seamlessly moved to their new location. This process lists steps in order:
+
+- A moratorium will be placed on Semantic Convention PRs to the specififcation
+  repository. (Caveat that PRs related to this proposal would be allowed).
+- Interactions between Semantic Conventions and the Specification will be
+  extracted such that the Specification can place requirements on Semantic
+  Conventions and *normative* specification language will remain in the
+  core specification directories.
+- A new repository `open-telemetry/semantic-conventions` will be constructed with
+  the proposed format and necessary `Makefile` / tooling.
+  - The new repository would be created by using `git filter-branch` to preserve
+    all existing semantic convention history. *This means all existing
+    semantic conventions will be in the new repository*.
+  - Github Actions, `Makefile` tooling and contributing / readmes would be
+    updated for the separate repository.
+  - **Note: At this point, the new location for semantic conventions should
+      be adoptable/usable.**
+- Semantic conventions in the Specification will be marked as moved with
+  links to the new location.
+  - The semconv YAML files in the specification repository *will be deleted*.
+  - All semconv markdown files will be updated such that:
+    - They no longer generate from YAML files.
+    - They include a header denoting deprecation and move to the new repository.
+- Instrumentation authors will update their code generation to pull from the new
+  semconv repository instead of the specification repository..
+- The new repository will have its first released versioned be `2.0.0`, to avoid
```

Comment:
> to avoid confusion with specification version

Not very clear on this motivation. What confusion? The intention is that these will diverge quickly anyway, and if 2.x SC vs. 1.x Spec looks like a difference, at some point Spec will also may be 2.x

Basically, I don't think we should bump to 2.x, just let the versions roll.

Reply:
My biggest concern here is that we don't go backward.  I don't know that I have a strong preference for starting from `1.22` (or whatever the next minor version would be at the time) or `2.0`, but starting at `1.0` or `0.1` would be problematic for versioned semconv packages and schema URLs.

Reply:
We definitely don't want to go backwards.

In addition @jsuereth suggests that we have a way to indicate in the version number that breaking changes occurred. I think it is not a bad idea, but we don't have to agree on it in this OTEP. It may be worth removing this particular sentence from the OTEP and make the OTEP strictly about the "separation from the spec".

Reply:
As discussed in Specification SiG -

Here's my reasoning for 2.0.

- We want to clearly denote that the semconv no longer follows the specification version.  SO the first release will definitely be different than the specification.
- We *cannot* go backwards in number, as SchemaURL requires semconv *ordering* of versions.
- I think it's important we use semantic versioning as an *implicit* signal of breaking changes to semconv, where the Schema file should give *explicit* signal of breaking changes.  We know HTTP semconv has broken for this next release.

Reply:
To resolve this, we moved this to "open questions".   We can continue this discussion of what version to use for the first release of semconv in the semconv repository.